### ISSUE TITLE
Question re: connection state during browser navigation

### ISSUE THREAD

**Issue by gtmadev** [emitter-io/emitter#288]
I have used another real-time framework where it manages connection state even when the app is not a SPA. For example, users are navigating from page to page and on each page load, a JS app is initializing, and subscribing to one or more channels. I noticed that on each page load with emitter, the connection id has changed. It doesn't seem to have a way to recover and use the same connection for that browser.

Is there a "best practices" for non SPA web apps that are going to connect over and over again as the user navigates throughout the app?

One idea I have is a very short message storage TTL (maybe 10 seconds) to account for user navigation. Upon navigation, the user connects again, subscribes, and gets any stored messages that are less than 10 seconds old, and compares them to a client-side list of messages it already knows about (using local storage). It's kind of a hack way to do it, but seems like it might work.

Basically.. we would replay the last 10 seconds of messages on each page load just to be sure we didn't miss something during navigation. Would something like that work?

**Comment by Florimond** [comment]
Indeed, nothing is stored on the server side that would allow a client to reconnect with the same id. There is no stickiness. 
It really depends on your use case, it could work, but it seems to me like relying any constant number seconds for the TTL is dangerous if not missing any message is crucial to you.
Do you have a completely serverless app? I suppose you don't have at any time some sort of server, or elected host that you can rely upon to give you a full state all at once...

**Comment by gtmadev** [comment]
I have prototyped the solution and it does work. I can vary the ttl if I need a little more confidence that a message won't be missed. The messages are not critical to the app working. The message arriving at the browser is more for user feedback about something being worked on in the background. If it is missed, it does not break anything, but the user feedback would not be seen (at least not immediately).

I see a connack function that is emitted on connection (in the docs). I looked at the source code for the JS project and I don't see the word clientId in the code at all. So, not sure what the docs are referring to. See below (from the docs)

**Event** `'connect'`

`function(connack) {}`

Emitted on successful (re)connection (i.e. connack rc=0).

- `connack` received connack packet. When `clean` connection option is `false` and server has a previous session for `clientId` connection option, then `connack.sessionPresent` flag is `true`. When that is the case, you may rely on stored session and prefer not to send subscribe commands for the client.
